pal/src/httpapi_sl.c

```diff
@@ -0,0 +1,492 @@
+// Copyright (c) Texas Instruments. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdlib.h>
+
+#include <ti/net/http/httpclient.h>
+
+#include "cert_sl.h"
+
+#include "azure_c_shared_utility/httpapi.h"
+#include "azure_c_shared_utility/strings.h"
+#include "azure_c_shared_utility/xlogging.h"
+
+#define CONTENT_BUF_LEN     (128 * 10)
+#define HTTP_SECURE_PORT    443
+#define HEADER_TO_STR(x) (headerFieldStr[(x) & (~HTTPClient_REQUEST_HEADER_MASK)])
+
+typedef struct {
+    HTTPClient_Handle cli;
+    char *prefixedHostName;
+    char *x509Certificate;
+    char *x509PrivateKey;
+    bool  isConnected;
+} HTTPAPI_Object;
+
+static const char * headerFieldStr[] = {
+"Age",
+"Allow",
+"Cache-Control",
+"Connection",
+"Content-Encoding",
+"Content-Language",
+"Content-Length",
+"Content-Location",
+"Content-Range",
+"Content-Type",
+"Date",
+"ETag",
+"Expires",
+"Last-Modified",
+"Location",
+"Proxy-Authenticate",
+"Retry-After",
+"Server",
+"Set-Cookie",
+"Trailer",
+"Transfer-Encoding",
+"Upgrade",
+"Vary",
+"Via",
+"Www-Authenticate",
+"Warning",
+"Accept",
+"Accept-Charset",
+"Accept-Encoding",
+"Accept-Language",
+"Authorization",
+"Cookie",
+"Expect",
+"Forwarded",
+"From",
+"Host",
+"If-Match",
+"If-Modified-Since",
+"If-None-Match",
+"If-Range",
+"If-Unmodified-Since",
+"Origin",
+"Proxy-Authorization",
+"Range",
+"TE",
+"User-Agent",
+};
```

Comment:
Is this definition necessary? It seems that the order of headers in this array must match the order expected by their `HTTPClient_getHeader()`, in which case I would expect that this array is defined elsewhere and available in a header file somewhere.